### PR TITLE
Backend modifications for pickles compatibility (rebased)

### PR DESCRIPTION
**Pull request by mimoo**
Same as https://github.com/o1-labs/proof-systems/pull/220 but rebased. Not sure if @imeckler is working on the other one so I avoided force pushing there.

### CODE REVIEW

File: circuits/kimchi/src/nolookup/constraints.rs [hunk 1]
```diff
@@ -605,7 +605,17 @@ impl<F: FftField + SquareRootField> ConstraintSystem<F> {
             // check if wires are connected
             for col in 0..PERMUTS {
                 let wire = gate.wires[col];
-                assert!(wire.col < PERMUTS);
+
+                if wire.col < PERMUTS {
```
Comment (mrmr1993) [inline circuits/kimchi/src/nolookup/constraints.rs]: This inequality is the opposite of what it should be

**Comment by mrmr1993** [comment]
Please don't rebase commits that have already been reviewed and/or referenced by other repos.

File: circuits/kimchi/src/nolookup/constraints.rs (hunk 1 above)
Comment (mimoo) [reply circuits/kimchi/src/nolookup/constraints.rs]: this is fixed in a later commit :o

**Comment by mimoo** [comment]
ah yeah that makes sense, I figured it would make reviews easier as they were a bunch of unrelated commits initially

**Comment by mimoo** [comment]
merged master into https://github.com/o1-labs/proof-systems/pull/220 instead

**Pull request closed without merging**